apps/web/lib/blog/repository/post-repository.ts

```diff
@@ -0,0 +1,47 @@
+import fs from "fs";
+import matter from "gray-matter";
+import path from "path";
+import { POSTS_DIRECTORY } from "../constants";
+import type { BlogFrontmatter } from "../types";
+
+export class PostRepository {
+  private postsDirectory: string;
+
+  constructor() {
+    this.postsDirectory = path.join(process.cwd(), POSTS_DIRECTORY);
+  }
+
+  getAllPostFiles(): string[] {
+    try {
+      return fs
+        .readdirSync(this.postsDirectory)
```

Comment:
I also think we should be avoiding the synchronous APIs, especially since there are updated async equivalents in `fs/promises` https://nodejs.org/api/fs.html#promises-api - these will actually run more efficiently on vercel anyway.